python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -1655,12 +1640,33 @@ def run(self, ccdExposure, *, dnlLUT=None, bias=None, deferredChargeCalib=None,
             isrFunctions.widenSaturationTrails(ccdExposure.getMaskedImage().getMask())
 
         # Brighter/Fatter
-        # FIXME: watch out (again) for gain units.
         # Units: electron
         if self.config.doBrighterFatter:
-            self.log.info("Applying Bright-Fatter kernels.")
+            self.log.info("Applying brighter-fatter correction.")
+
             bfKernelOut, bfGains = self.getBrighterFatterKernel(detector, bfKernel)
-            ccdExposure = self.applyBrighterFatterCorrection(ccdExposure, flat, dark, bfKernelOut, bfGains)
+
+            # Needs to be done in electrons; applyBrighterFatterCorrection
+            # will convert the image if necessary.
+            if exposureMetadata["LSST ISR UNITS"] == "electron":
+                brighterFatterApplyGain = False
+            else:
+                brighterFatterApplyGain = True
+                exposureMetadata["LSST ISR UNITS"] == "electron"
```

Comment:
This is not correct.  The gain is applied in a context, so the application is temporary.  We don't want to output `electron` units if `doApplyGain = False`.

Reply:
(And to be fair, this will be a very unusual code path, but I want to make sure it's correct).

Reply:
Ah! I forgot that the application of gain in the helper function applies gain through gainContext